presto/dbapi.py

```diff
@@ -281,17 +282,46 @@ def _prepare_statement(self, operation, statement_name):
 
         raise presto.exceptions.FailedToObtainAddedPrepareHeader
 
+
     def _get_added_prepare_statement_presto_query(
         self,
         statement_name,
         params
     ):
-        sql = 'EXECUTE ' + statement_name + ' USING ' + ','.join(map(str, params))
+        sql = 'EXECUTE ' + statement_name + ' USING ' + ','.join(map(self._format_prepared_param, params))
 
         # No need to deepcopy _request here because this is the actual request
         # operation
         return presto.client.PrestoQuery(self._request, sql=sql)
 
+    def _format_prepared_param(self, param):
+        """
+        Formats parameters to be passed in an
+        EXECUTE statement.
+        """
+        if isinstance(param, bool):
+            return "true" if param else "false"
+        if isinstance(param, float):
+            return "DOUBLE '%s'" % param
+        if isinstance(param, str):
+            return ("'%s'" % param.replace("'", "''"))
+        if isinstance(param, bytes):
+            return "X'%s'" % param.hex()
+        if isinstance(param, datetime.datetime):
+            datetime_str = param.strftime("%Y-%m-%d %H:%M:%S.%f %Z").rstrip(" ")
+            return "TIMESTAMP '%s'" % re.sub(r'\.([0-9]{3})([0-9]{3})', r'.\1', datetime_str)
+        if isinstance(param, list):
+            return "ARRAY[%s]" % ','.join(map(self._format_prepared_param, param))
+        if isinstance(param, dict):
+            keys = self._format_prepared_param(list(param.keys()))
+            values = self._format_prepared_param(list(param.values()))
+            return "MAP(%s, %s)" % (keys, values)
+        if isinstance(param, uuid.UUID):
+            return "UUID '%s'" % str(param)
+
+        return str(param)
```

Comment:
thought an exception might be appropriate here, rather than presto spitting out a random syntax error. Thoughts?

Reply:
Agreed.

Reply:
added, used `NotSupportedError` as the exception. Let me know if that is the correct one to use  here.